service/collector.go

```diff
@@ -214,6 +214,12 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 		return fmt.Errorf("invalid configuration: %w", err)
 	}
 
+	if col.logger, err = telemetrylogs.NewLogger(cfg.Service.Telemetry.Logs, col.set.LoggingOptions); err != nil {
+		return fmt.Errorf("failed to get logger: %w", err)
```

Comment:
Anything against using `errors.Wrap`?

Reply:
Extra dependency and go recommends this way. Will work with Is/As